### PR TITLE
Helper class to translate platform BOM coordinates to the corresponding JSON catalog coordinates

### PR DESCRIPTION
**Pull request by aloubyansky**
At this point it is mainly intended for the external dev tools clients but this class will certainly be used internally as well.

### CODE REVIEW

**Comment by ia3andy** [comment]
Maybe a few tests to make sure it works as expected?

**Comment by aloubyansky** [comment]
Certainly a good idea :) Will add them.

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package io.quarkus.registry.util;
+
+import io.quarkus.maven.ArtifactCoords;
+import io.quarkus.registry.Constants;
+
+public class PlatformArtifacts {
+
+    public static ArtifactCoords bomToCatalog(ArtifactCoords bom) {
```
Comment (ia3andy) [inline independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java:8]:
```suggestion
    public static ArtifactCoords getCatalogCoordsFromBom(ArtifactCoords bom) {
```

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        return bomToCatalog(bom.getGroupId(), bom.getArtifactId(), bom.getVersion());
+    }
+
+    public static ArtifactCoords bomToCatalog(String bomGroupId, String bomArtifactId, String bomVersion) {
```
Comment (ia3andy) [inline independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java:12]:
```suggestion
    public static ArtifactCoords getCatalogCoordsFromBom(String bomGroupId, String bomArtifactId, String bomVersion) {
```

Comment (aloubyansky) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java:12]: If we do that, shouldn't we go all the way? PlatformArtifacts.getCatalogArtifactFromBomArtifact(...)?

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        return new ArtifactCoords(bomGroupId, bomArtifactId + Constants.PLATFORM_DESCRIPTOR_ARTIFACT_ID_SUFFIX, bomVersion,
+                "json", bomVersion);
+    }
+
+    public static ArtifactCoords toCatalog(ArtifactCoords coords) {
```
Comment (ia3andy) [inline independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java:17]: I don't really see what this one bring over the `bomToCatalog`?

Comment (aloubyansky) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java:17]: bomToCatalog doesn't expects the BOM coords as the argument. This method is used when you don't know whether the artifact you have is a catalog one or the bom. There are such cases in the codebase.

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java (hunk 2 above)
Comment (ia3andy) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java:12]: I wouldn't mind, as soon as it's descriptive enough. 

My point is that with `bomToCatalog` you need to look at the signature and the code to understand what it does.

File: independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java (hunk 3 above)
Comment (ia3andy) [reply independent-projects/tools/registry-client/src/main/java/io/quarkus/registry/util/PlatformArtifacts.java:17]: ah ok sure so `ensureCatalogArtifact` maybe?

**Review by ia3andy — APPROVED** [review]
Added some suggestions but it's fine already ;)

**Comment by aloubyansky** [comment]
Done, thanks.

**Review by ia3andy — APPROVED** [review]
Cool :)

**Review by gastaldi — APPROVED** [review]
LGTM, but I wouldn't mind having some Javadoc for the methods and the class :)

**Pull request merged**